mantidproject/mantid
# 35442 quick bayes

**Description of work**
The quasielasticbayes package currently used in Mantid relies on old unmaintained fortran code. This PR adds code to a new python library that can be used instead. Once this has been validated by scientists we can remove the old algorithms. 

The installation is a hack. The plan is to add the package to conda forge https://github.com/mantidproject/quickBayes/issues/66 and we can then update the installation https://github.com/mantidproject/mantid/issues/35918
The hack was chosen due to time constraints and its temporary nature. 

The old algorithms will be deprecated in https://github.com/mantidproject/mantid/issues/35917

**To test:**
Load a "_red" and a "_res" file (I can send them)

use the bayesquasi2 algorithm with Ql mode, start = -0.4 and end =0.4, elastic true
Check that the output gives some nice fits (1, 2 and 3 lorentzians)

use the bayesquasi2 algorithm with QSe mode, start = -0.4 and end =0.4, elastic true
Check that the output gives a nice fit (1 stretched exp)

use the bayesstretch2 algorithm, start = -0.4 and end =0.4, elastic true
The default beta and FWHM values are ok, but you may want to change the number of each. However, the code can be slow a 10x10 grid takes about 5 min on my machine
Check the contours (I find the wireframes the best plot) show a nice peak


*There is no associated issue.*


Added release notes


---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
